### PR TITLE
[occm] OpenStack Cloud Controller Manager Helm Chart

### PR DESCRIPTION
**Pull request by eumel8**
**What this PR does / why we need it**:

Adding Helm Chart for occm deployment

**Which issue this PR fixes(if applicable)**:
none

**Special notes for reviewers**:

none

**Release note**:
```release-note
OpenStack Cloud Controller Manager Helm Chart
```

### CODE REVIEW

File: charts/openstack-cloud-controller-manager/values.yaml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+# tolerations:
+#   - key: node.cloudprovider.kubernetes.io/uninitialized
+#     value: "true"
+#     effect: NoSchedule
+#   - key: node-role.kubernetes.io/controlplane
+#     value: "true"
+#     effect: NoSchedule
+#   - key: node-role.kubernetes.io/etcd
+#     value: "true"
+#     effect: NoExecute
```
Comment (morremeyer) [inline charts/openstack-cloud-controller-manager/values.yaml:31]: I’d make the commented tolerations the default as I think this is the most common setup and will facilitate configuration for the users.

**Review by morremeyer — CHANGES_REQUESTED** [review]
Looks good to me for the most part, I PRd some changes (as @eumel8 and I discussed on Slack) to @eumel8's fork directly.

For the part here, there’s only the one change I’d make to the default tolerations.

**Review by lingxiankong — CHANGES_REQUESTED** [review]
The helm chart is in good shape, besides the comment I've left, can we add more comments for the options in the `values.yaml`? e.g. the default value, what if it's set to true, etc.?

File: charts/openstack-cloud-controller-manager/values.yaml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+controllerExtraArgs: {}
+# controllerExtraArgs: |-
+#   - --cluster-name=my-cluster
+
+loadbalancerService: {}
```
[the remark this answers is not in the document]
Comment (eumel8) [reply charts/openstack-cloud-controller-manager/values.yaml:50]: I know, this have disadvantages and dependencies, on the other hand it`s a main functionality of occm. Would you suggest an extra Chart only with the service loadbalancer?

[the remark this answers is not in the document]
Comment (morremeyer) [reply charts/openstack-cloud-controller-manager/values.yaml:50]: I agree with @lingxiankong here. The occm should chart should only deploy the occm itself.

services of type LoadBalancers etc. should be created by the applications that need it in their respective charts.

[the remark this answers is not in the document]
Comment (eumel8) [reply charts/openstack-cloud-controller-manager/values.yaml]: ok,  moved to https://github.com/kubernetes/cloud-provider-openstack/pull/1466

**Comment by lingxiankong** [comment]
/lgtm

Good to have. Thanks!

**Comment by jichenjc** [comment]
/approve

thanks

**Pull request merged**